services/spar/test-integration/Test/Spar/Scim/UserSpec.hs

```diff
@@ -493,13 +493,54 @@ specDeleteUser = do
                 !!! const 405 === statusCode
 
     describe "DELETE /Users/:id" $ do
+        it "should be delete successfully and be idempotent" $ do
+            (tok, _) <- registerIdPAndScimToken
+            user <- randomScimUser
+            storedUser <- createUser tok user
+            let uid = scimUserId storedUser
+
+            spar <- view teSpar
+            deleteUser_ (Just tok) (Just uid) spar
+                !!! const 204 === statusCode
+            deleteUser_ (Just tok) (Just uid) spar
```

Comment:
I believe that it should return 404 the second time.

> Service providers MAY choose not to permanently delete the resource but MUST return a 404 (Not Found) error code for all operations associated with the previously deleted resource.

Reply:
https://github.com/wireapp/hscim/issues/27

To be able to move forward, I'm also going to fix this here without touching hscim.